processing/src/main/java/org/apache/druid/segment/serde/NestedCommonFormatColumnPartSerde.java

```diff
@@ -308,6 +308,10 @@ public void read(ByteBuffer buffer, ColumnBuilder builder, ColumnConfig columnCo
       ColumnType logicalType = simpleType == null ? ColumnType.NESTED_DATA : simpleType;
       builder.setType(logicalType);
       builder.setNestedCommonFormatColumnSupplier(supplier);
+      // nested columns only have a null value index
+      if (hasNulls) {
+        builder.setIndexSupplier(supplier, false, false);
+      }
```

Comment:
I wonder if it shouldn't be another class that depends on the `supplier` instead of just the supplier itself.  Specifically, I think that if the index supplier ends up needing to track resources to like decompress something or whatever, we will have lifecycle management issues.  The `Column` itself operates in a managed lifecycle, so it might actually be better to move this `IndexSupplier` to be expected to exist on the `Column` itself, which might simplify reuse as well?

Definitely a larger change than what you have here though.

Reply:
yea, we've talked about doing this in the past, i agree it would be nicer to move it to the same place we get selectors from, but we need to re-arrange some other stuff to make this stuff closable and ensure it gets closed.